api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java

```diff
@@ -9,8 +22,54 @@
 @Service
 public class MailServiceImpl implements MailService {
 
+    private final Provider<MandrillApi> mandrillApiProvider;
+    private final Provider<CloudStorageService> cloudStorageServiceProvider;
+    private Provider<WorkbenchConfig> workbenchConfigProvider;
+
+    @Autowired
+    public MailServiceImpl(Provider<MandrillApi> mandrillApiProvider,
+                           Provider<CloudStorageService> cloudStorageServiceProvider,
+                           Provider<WorkbenchConfig> workbenchConfigProvider) {
+        this.mandrillApiProvider = mandrillApiProvider;
+        this.cloudStorageServiceProvider = cloudStorageServiceProvider;
+        this.workbenchConfigProvider = workbenchConfigProvider;
+    }
+
     public void send(Message msg) throws MessagingException {
         Transport.send(msg);
     }
 
+    public void sendWelcomeEmail(String contactEmail, String password, User user) throws MessagingException {
+        String apiKey = cloudStorageServiceProvider.get().readMandrillApiKey();
+        MandrillApiKeyAndMessage keyAndMessage = new MandrillApiKeyAndMessage();
+        keyAndMessage.setKey(apiKey);
+        MandrillMessage msg = buildWelcomeMessage(contactEmail, password, user);
+        keyAndMessage.setMessage(msg);
+        try {
+            MandrillMessageStatuses msgStatuses = mandrillApiProvider.get().send(keyAndMessage);
+            for (MandrillMessageStatus msgStatus : msgStatuses) {
+                if (msgStatus.getRejectReason() != null) {
+                  throw new MessagingException(msgStatus.getRejectReason());
```

Comment:
What are the possible reject reasons?

Reply:
"hard-bounce", "soft-bounce", "spam", "unsub", "custom", "invalid-sender", "invalid", "test-mode-limit", "unsigned", or "rule". But given these are strings set by Mandrill, figured it was best not to "hard code" these in. But in the JSON response from them there's a field "reject_reason" that if it was rejected, the reason is here. Otherwise it's null.